mlbench_core/evaluation/pytorch/metrics.py

```diff
@@ -102,4 +95,87 @@ def __call__(self, loss, output, target):
         Returns:
             float
         """
-        return torch.exp(loss)
\ No newline at end of file
+        return torch.exp(loss)
+
+
+class DiceCoefficient(MLBenchMetric):
+
+    def __call__(self, loss, output, target):
+        """ Computes the Dice Coefficient of a Binary classification problem
+
+        Args:
+            loss (:obj:`torch.Tensor`): Not Used
+            output (:obj:`torch.Tensor`): Output of model
+            target (:obj:`torch.Tensor`): Target labels
+
+        Returns:
+            float: Dice Coefficient in [0,1]
+        """
+        eps = 0.0001
+        output, target = output.float(), target.float()
+        self.inter = torch.dot(output.view(-1), target.view(-1))
+        self.union = torch.sum(output) + torch.sum(target) + eps
+
+        t = (2 * self.inter.float() + eps) / self.union.float()
+        return t
+
+    @property
+    def name(self):
+        """str: Name of this metric"""
+        return "Dice Coefficient"
+
+
+class F1Score(MLBenchMetric):
+    def __init__(self, threshold=0.5, eps=1e-9):
+        """ F1-Score metric
+
+        Args:
+            threshold (float): Threshold for prediction probability
+        """
+        super(F1Score, self).__init__()
+        self.threshold = threshold
+        self.eps = eps
+
+    def __call__(self, loss, output, target):
+        """ Computes the F1-Score of a Binary classification problem
+
+        Args:
+            loss (:obj:`torch.Tensor`): Not Used
+            output (:obj:`torch.Tensor`): Output of model
+            target (:obj:`torch.Tensor`): Target labels
+
+        Returns:
+            float: F1-Score in [0,1]
+        """
+
+        y_pred = torch.ge(output.float(), self.threshold).float()
+        y_true = target.float()
+
+        true_positive = (y_pred * y_true).sum(dim=0)
+        precision = true_positive.div(y_pred.sum(dim=0).add(self.eps))
+        recall = true_positive.div(y_true.sum(dim=0).add(self.eps))
+
+        return torch.mean(
+            (precision * recall).div(precision + recall + self.eps).mul(2))
+
+    @property
+    def name(self):
+        return "F1-Score"
+
+
+class Accuracy(MLBenchMetric):
```

Comment:
this is the same as `TopKAccuracy(k=1)`, no?

Reply:
I wasn't sure how TopKAccuracy works. This computes the number of correct classifications over the total number of classifications and outputs

Reply:
It is in fact the same, I will delete my class and use TopKAccuracy(k=1)

Reply:
TopKAccuracy is number of correct predictions in the top k (highest confidence) outputs of the models. E.g. for a model that predicts the probability of classes `car`, `plane` and `boat` (with a softmax) and a predictions of (0.5, 0.3 and 0.2 respectively),if the highest prediction is for `car`, and the second highest is for `plane` and the correct class is `plane`, Top1Accuracy would count it as wrong, but Top2Accuracy would count it as a correct prediction (since it's in the 2 highest predictions). It's common in some ML tasks (especially language modeling) to check if the correct prediction is somewhere in the top 5 labels/targets/outputs with the highest confidence to evaluate a model, as the highest confidence prediction could be `computer scientist` but the correct label would be `programmer`, both of which are technically correct for predicting next ngram in a sentence.

So Top1Accuracy reduces to Accuracy. I hope that makes sense.

Alternatively, `Accuracy` could inherit from `TopKAccuracy` and use a fixed `k=1` in its initialization.

Reply:
I actually tested it with some dummy values, and it seems Top1Accuracy = 1 - Accuracy:
```
    output = torch.tensor([1, 1, 0, 1, 0]).reshape(5, 1)
    target = torch.tensor([0, 1, 0, 0, 1]).reshape(5, 1)

    top = TopKAccuracy(topk=1)
    acc = Accuracy()
    expected_score = 2 / 5

    acc_score = acc(None, output, target)
    actual_score = top(None, output, target)
    assert acc_score == actual_score  =====> Fails because 0.4 != 0.6
```

Reply:
```
output = torch.tensor([1, 1, 0, 1, 0]).reshape(5, 1)
target = torch.tensor([0, 1, 0, 0, 1]).reshape(5, 1)
```

should be something like
```
output = torch.tensor([[0,1], [0,1], [1,0], [0,1], [1,0]]).reshape(2, 5)
target = torch.tensor([0, 1, 0, 0, 1]).reshape(5, 1)
```

with 2 classes (true/false) the prediction is `[probability_of_false, probability_of_true]`, so something like [0.78, 0.22] (after softmax) means the network gives 78% probability that it's false and 22% probability that it's true.

Reply:
I see, but the model outputs a single probability, not two (the probability of being 1)